lib/libefi/rdwr_efi.c

```diff
@@ -1255,19 +1300,61 @@ efi_use_whole_disk(int fd)
 	efi_label->efi_parts[resv_index].p_start += difference;
 	efi_label->efi_last_u_lba = efi_label->efi_last_lba - nblocks;
 
-	rval = efi_write(fd, efi_label);
-	if (rval < 0) {
-		if (efi_debug) {
-			(void) fprintf(stderr,
-			    "efi_use_whole_disk:fail to write label, rval=%d\n",
-			    rval);
+#ifdef BLKPG_RESIZE_PARTITION
+	/*
+	 * Delete the reserved partition since we're about to expand
+	 * the data partition and it would overlap with the reserved
+	 * partition.
+	 */
+	rval = call_blkpg_ioctl(fd, BLKPG_DEL_PARTITION, 0, 0, resv_index + 1);
```

Comment:
Are we adding the `+1` here and below because `efi_alloc_and_read()` returns an array that starts from 0 but the partitions start from 1?

Reply:
That's correct, the efi logic numbers the partitions starting with 0 but the BLKPG ioctl starts with 1.